hil/client/node.py

```diff
@@ -32,7 +32,26 @@ def register(self, node, subtype, *args):
         # and currently active drivers for HIL
         # obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
         # obm_types = ["ipmi", "mock"]
-        raise NotImplementedError
+        obm_api = "http://schema.massopencloud.org/haas/v0/obm/"
+        ext_url = self.object_url('active_extensions')
+        # This is a temp fix. obmd will let node_register no longer
+        # need a type so a new design will be required.
+        obm_types = ["ipmi", "mock"]
+        
+        if subtype in obm_types:
+            if len(args) == 3:
+                obminfo = {"type": obm_api + subtype, "host": args[0],
+                           "user": args[1], "password": args[2]
+                           }
+        else:
+            raise Exception('ERROR: subtype ' + subtype +
```

Comment:
I am not sure whether it is worth to write a new exception in the base.py for this function. 
1. After having the `obmd`, are we still checking the length of args?
2. Would this function be the only function that uses the new exception? 

@zenhack @naved001 thoughts?

Reply:
Switch register could also use this. And when we have some other network backend then the call to create networks might require different number of arguments (not sure on this though).

Reply:
Probably best not to just raise `Exception`; this makes it hard to just handle the relevant exception. I would make another class.